docs/agent-images.md

```diff
@@ -0,0 +1,203 @@
+# Agent images
+
+Agents in Rover run in isolated environments, as much as possible. The
+idea behind this strategy is that the agent can make as many changes
+as it needs in order to improve, implement, build, check, test... on a
+given project, without impacting the host machine.
+
+When Rover runs, it is possible to choose what image will be used to
+spawn the agent in. This gives flexibility to have specific OS's
+(e.g. Alpine, Debian, Ubuntu), as well as some core tooling already
+installed for the project, what will increase development speed,
+because the agent won't need to install new packages.
+
+## Rover provided agent images
+
+### Development images
+
+The code at the `main` branch points to
+`ghcr.io/endorhq/rover/agent-dev:latest`. Automation builds and pushes
+to this image when a new commit is pushed to the `main` branch.
+
+### Tagged images
+
+Whenever we create a new tag in Rover, a new image will be pushed with
+that tag name at: `ghcr.io/endorhq/rover/agent:<tag>`.
```

Comment:
I like this idea, but I miss some details about how we these images are built (I know it will be in the future, but we can still document it already).

For example, currently the git tags are created after we bump the version and commit the change in a `GitHub` action. What would be the process with this new image system?